src/apps/vpn/mozillavpn.cpp

```diff
@@ -422,8 +420,6 @@ void MozillaVPN::maybeStateMain() {
     return;
   }
 
-  Q_ASSERT(m_private->m_serverData.hasServerData());
```

Comment:
Why removing this? We must have a server selection before showing the main view. It seems that this is not guaranteed in this PR. 

I wonder if we should combine the serverlist task and this one in 1 single operation if the server data is empty.

We probably need a TaskSequence to wrap TaskServer and this new task, as part of the group of tasks to run during the authentication. 

And maybe we should keep this pickRandom if we arrive here without selection. 

Thought?

Reply:
I kinda figured I this would have gotten flagged.

Yeah, I tried a few things on this one since we now do the server selection async, and there were a few things that came to mind:
- Add a call to get the geoIP as a part of the periodic refresh and cache it somewhere. Then server selection can be done synchronously, like with the `pickRandom()`. But then the geoIP gets very muddled once the user starts changing their location by activating the VPN. But this isn't really a big concern, and the GeoIP is already inaccurate if the VPN was active when launching the UI anyways.
- Schedule server selection as part of the periodic refresh, but just do a no-op if the server is already set, but this would work against my goal of adding an automatic server option.
- If the user does manage to get through the activation screens fast enough to arrive at the main screen before server selection, it doesn't actually break anything... they just get an empty server, and with [5812cf2](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5398/commits/5812cf2d9028edfb22a4c80d67324f398ecf8503) it just picks a server upon activation.
- My long term evil plan is to add an `Automatic` server option, which runs this task upon every activation in which case we can just leave the server unset intentionally via settings.

Ultimately, I felt that simply removing the assert was the cleanest option.

Reply:
After playing around a bit with an automatic server option, it seems that we kind of need a `Location` model anyways. So it actually better to put this assert back and deal with this another way.